screens/visualization/sketch.py

```diff
@@ -104,16 +103,32 @@ def __call__(self):
         return results
 
 
-if __name__ == '__main__':
-    if len(sys.argv) < 3:
-        figures = [None] if len(sys.argv) <= 1 else sys.argv[1]
-        filename = None
-        plt.ion()
-    else:
-        filename = sys.argv[1]
-        figures = sys.argv[2:]
-        plt.ioff()
-
+def sketch(th, th_scale=2., screens=True, direct=True, velocity=True,
+           scales=False, filename=None):
+    """Draw a schematic of the thin-screen model of for scintillation.
+
+    Parameters
+    ----------
+    th : `~astropy.units.Quantity`
+        Angles of the scattering images from the core image.
+    th_scale : float, optional
+        Scalar to rescale angles, so the scattering points fit vertically
+        in the sketch figure.
+    screens : bool, optional
+        Whether or not to draw the scintillation screen.
+    direct : bool, optional
+        Whether or not to draw direct line-or-sight beam.
+    velocity : bool, optional
+        Whether or not to include an arrow to indicate the pulsar's motion.
+    scales : bool, optional
+        Whether or not to include scale bars indicating physical sizes.
+    filename : string, optional
+        Path to output file.
+    """
+
+    screen_y_from_th = th_scale * np.delete(th.value, np.where(th == 0))

```

Comment:
Why is this helpful? I think it should be up to the person using the function to remove any zeros, and also to pass in a scale that makes sense.

I do like the idea of making sure `theta` are actual angles, but would suggest to make the scale a proper stretch in the `y` direction, i.e., calculate a y from `screen_x * np.tan(theta) * screen_y_scale`.

Reply:
Removing the zero was meant to avoid drawing the direct line of sight twice (once from the normal 'screens', once from 'direct') with different linestyles. I updated this, so it only gets removed when 'direct' is active and followed your other suggestions here.